### ISSUE TITLE
For auto-detection: specify folder(s) to watch

### ISSUE THREAD

**Issue by laawn** [qawatake/obsidian-binary-file-manager-plugin#6]
For auto-detection: it'd be great if we could have the plugin monitor specified folders for new attachments/files rather than simply watching the entire vault. Ideally, when selecting the radio button for "Auto-Detection," a folder selection option would open below this, similar to the one for "New file location."

**Comment by Ravenclawer** [comment]
I thought it would be better to have this function to watch folders outside the vault for better integration with the windows file explorer.

**Comment by ThadySenior** [comment]
I agree with OP - keen to use this automation for automatically creating embeds for supported filetypes added to an 'import' or 'scans' folder but not for any documents added (e.g. if I were to drop an image into a note). Might also be nice if it were possible to set up different templates / treatments for different monitored folders (e.g. if file dropped into 'scans' folder to use a different template from if dropped into 'import' folder).

**Comment by sbj175** [comment]
Also agree with OP!
While watching a folder outside the vault might have some advantages, I doubt that it's possible for an Obsidian plugin as Obsidian itself doesn't know anything about what's going on outside the vault. Having said that, watching just a specific folder within the vault would be *awesome*.

**Comment by thomasotterdal** [comment]
+1 for this! Found great use for this plugin as an import folder for academic notes on e-ink, but would be great to only watch the folder for those notes. Thanks!

**Comment by willjasen** [comment]
Heya everyone, I'm day 1 new to Obsidian but I've been developing my workflow for the last 12+ hours or so and found a need for watching a folder. I noticed that this repo hasn't been updated in about two years but thankfully @matthewturk gave me a start for this feature. My [version 0.4.0 release](https://github.com/willjasen/obsidian-binary-file-manager-plugin/tree/0.4.0) watches only the root of a specified folder (subfolders are not watched) and binary files are moved into an attachments folder after its metadata is created. The workflow idea here is to have notes in a particular folder but all attachments hidden away in a subfolder - e.g. - scan PDFs/images from an image scanner that end up in the watched folder but are then properly replaced with a note that can be added onto.

The code I added isn't pretty and may contain bugs but it's working for my needs so far - pretty good for my first day of doing TypeScript! I hope someone finds this useful! :smiley:

**Comment by jebeaudoin** [comment]
@willjasen Thanks for doing this. I'll give it a try.

**Comment by guix54** [comment]
hello, Thanks for the job, how I can specify the folder to watch?
perhaps my install is not clean...i have just put the repo into the plugins folder..

**Comment by willjasen** [comment]
Because this is a custom development, you have to build with npm and move the resulting output into the .obsidian/plugins folder, then go into Obsidian and enable the plugin. I'm away from my computer at the moment, but I'll show an example in a bit.

**Comment by willjasen** [comment]
I have committed [my build script ](https://github.com/willjasen/obsidian-binary-file-manager-plugin/blob/dev/build.sh )into the dev branch of my repo. You'll want to change the SOURCE_DIR and PLUGIN_DIR variables to your paths. After the build, disable/enable the plugin in Obsidian. If the plugin is successfully installed, it should show up like below:

<img width="719" alt="Screenshot 2024-01-01 at 8 03 49 PM" src="https://github.com/qawatake/obsidian-binary-file-manager-plugin/assets/613571/939b0cf9-237d-4130-8a45-515f1eb84216">
